app/controllers/cards_controller.rb

```diff
@@ -1,6 +1,43 @@
 class CardsController < ApplicationController
   
-  def index   
+  def index
+    @cards = Card.all
   end
 
+  def show
+    @card = Card.find(params[:id])
+  end
+
+  def new
+    @card = Card.new  
+  end
+
+  def create
+    @card = Card.new(card_params) 
+
+    if @card.save
+      redirect_to @card, :notice => 'The card is saved'
+    else
+      render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.